includes/oauth/class-google-oauth.php

```diff
@@ -130,7 +130,13 @@ private static function save_auth_credentials( $tokens ) {
 
 		if ( $tokens['csrf_token'] !== $saved_csrf_token ) {
 			Logger::log( 'Failed saving credentials - CSRF token mismatch.' );
-			return new \WP_Error( 'newspack_google_oauth', __( 'Session token mismatch.', 'newspack' ) );
+			return new \WP_Error(
+				'newspack_google_oauth',
+				__( 'Session token mismatch.', 'newspack' ),
+				[
+					'status' => 403,
+				]
+			);
 		}
 
 		$auth                 = self::get_google_auth_saved_data();
```

Comment:
We could take advantage of more checks here, to make sure that the incoming `$tokens` data is valid.

Reply:
Good idea – removed the refresh token optionality, too. In 53c173178a8f6fe071131167c2aca1f3a0ddcd6b

Reply:
That's right, reverted in a540601e33509e95fec6c5a0e9121d4aaa1f55d4